front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -139,9 +146,29 @@ const SpaceTimeChartWrapper = ({
   const [draggingState, setDraggingState] = useState<DraggingState>();
 
   const isTimetableItemValid = useMemo(() => {
+    let selectedItemId: TrainScheduleId | PacedTrainId | OccurrenceId = selectedProjectionId;
+
+    if (isOccurrenceId(selectedProjectionId)) {
+      selectedItemId = extractPacedTrainIdFromOccurrenceId(selectedProjectionId);
+    }
```

Comment:
If you write it like this, ts will know that `selectedItemId` is a `TimetableItemId` (`PacedTrainId` or `trainScheduleId`)
```suggestion
    const selectedItemId = isOccurrenceId(selectedProjectionId)
      ? extractPacedTrainIdFromOccurrenceId(selectedProjectionId)
      : selectedProjectionId;
```

Reply:
Done